frame/atomic-swap/src/lib.rs

```diff
@@ -35,37 +35,104 @@ use codec::{Encode, Decode};
 use sp_runtime::RuntimeDebug;
 
 /// Pending atomic swap operation.
-#[derive(Clone, RuntimeDebug, Eq, PartialEq, Encode, Decode)]
-pub struct PendingSwap<AccountId, Balance, BlockNumber> {
+#[derive(Clone, Eq, PartialEq, RuntimeDebug, Encode, Decode)]
+pub struct PendingSwap<T: Trait> {
 	/// Source of the swap.
-	pub source: AccountId,
-	/// Balance value of the swap.
-	pub balance: Balance,
+	pub source: AccountIdFor<T>,
+	/// Action of this swap.
+	pub action: T::SwapAction,
 	/// End block of the lock.
-	pub end_block: BlockNumber,
+	pub end_block: BlockNumberFor<T>,
 }
 
-/// Balance type from the pallet's point of view.
-pub type BalanceFor<T> = <<T as Trait>::Currency as Currency<<T as frame_system::Trait>::AccountId>>::Balance;
-
 /// AccountId type from the pallet's point of view.
 pub type AccountIdFor<T> = <T as frame_system::Trait>::AccountId;
 
 /// BlockNumber type from the pallet's point of view.
 pub type BlockNumberFor<T> = <T as frame_system::Trait>::BlockNumber;
 
```

Comment:
These type aliases aren't really needed. Should be easy enough to always refer to them as `T::AccountId` or `T::BlockNumber`, but its all the same I guess.